semgrep-core/src/analyzing/Dataflow_svalue.ml

```diff
@@ -515,9 +524,18 @@ let transfer :
               let ccall = sym_prop instr.iorig in
               update_env_with inp' var ccall
         | CallSpecial
-            (Some { base = Var var; rev_offset = [] }, (Concat, _), args) ->
-            (* var = concat(args) *)
-            let cexp = eval_concat inp' args in
+            (Some { base = Var var; rev_offset = [] }, (special, _), args) ->
+            let cexp =
+              (* We try to evalaute the special function. *)
```

Comment:
```suggestion
              (* We try to evaluate the special function. *)
```

Reply:
I think something like `Depending on what the special function is, we evaluate it` might be even more clear? It's ok though. Also, just to be clear, in the case of Concat I assume we will propagate the evaluated value?

Reply:
Thanks for asking these questions, I clarified the comment. I wrote:

> We try to evaluate the special function, if we know how. If we don't know how to evaluate this function, or if the evaluation fails, then we do sym-prop.

So if evaluating `Concat` gives a constant then yes we propagate that, but if we fail to get a constant from `Concat` then we do sym-prop.